lib/v2/screens/dashboard/token_cards_widget.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:carousel_slider/carousel_controller.dart';
+import 'package:carousel_slider/carousel_options.dart';
+import 'package:carousel_slider/carousel_slider.dart';
+import 'package:dots_indicator/dots_indicator.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/v2/constants/app_colors.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/token_balances_bloc.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/token_balances_event.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/token_balances_state.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/wallet_bloc.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/wallet_state.dart';
+
+import 'components/currency_info_card_widget.dart';
+
+class TokenCardsWidget extends StatelessWidget {
+  final CarouselController _controller = CarouselController();
+
+  TokenCardsWidget({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+        create: (_) => TokenBalancesBloc()..add(const OnLoadTokenBalances()),
+        child: BlocListener<WalletBloc, WalletState>(
+          listenWhen: (context, state) => state.pageState == PageState.loading,
+          listener: (context, state) {
+            BlocProvider.of<TokenBalancesBloc>(context).add(const OnLoadTokenBalances());
+          },
```

Comment:
What we want to accomplish here?

Refesh token balances rigth?

My question how often do we want to it?

The current behaviour (without this logic) is 

Every time WalletScreen (the parent of TokensCards) rebuilds also TokenCards it does so WalletScreen ans TokensCards both of them fires their first event when they are mounted ( OnLoadSomething ).
_**You can test this by set break points in the first instruction below their build function or in how load events appears in the logs._

From the previous behavior I can say that the logic of this listener is not necessary.

Reply:
When main screen is loading, we also want to be loading

Reply:
This is for the "refresh" indicator

Reply:
I tested, without this code the balances do not update when the user "pulls to refresh" on the main screen

listenerWhen could be removed, but might lead to additional updates later when the wallet has more states, etc.

Reply:
You are rigth that was because wallet I was not using the BlocBuilder and the refresh indicator does not trigger a rebuild.

now this TokensCards has const contructor it means no rebulds when Wallet's BlocBulilder fires.

So you approach is here is correct :)

Reply:
I already made some structure changes and fixed a bug the event 
` BlocProvider.of<WalletBloc>(context)..add(const RefreshDataEvent());`

It was throwing an error of no context from where to get the WalletBloc. That is why I added the BlocBuilder.

Reply:
I fixed that but in another PR... sorry ;)